libs/velo-external-db-core/src/service/data.ts

```diff
@@ -32,37 +31,51 @@ export default class DataService {
 
     async insert(collectionName: string, item: Item, fields?: ResponseField[]) {
         const resp = await this.bulkInsert(collectionName, [item], fields)
-        return { item: asWixData(resp.items[0]) }
+        return { item: resp.items[0] }
     }
 
     async bulkUpsert(collectionName: string, items: Item[], fields?: ResponseField[]) {
         await this.storage.insert(collectionName, items, fields, true)
         return { items: items.map( asWixData ) }
     }
 
-    async bulkInsert(collectionName: string, items: Item[], fields?: ResponseField[]) {
-        await this.storage.insert(collectionName, items, fields)
-        return { items: items.map( asWixData ) }
+    async bulkInsert(collectionName: string, _items: Item[], fields?: ResponseField[]) {
+        const items = await Promise.all((_items.map( item => this.storage.insert(collectionName, [item], fields)
+                                                                          .then(_i => asWixData(item))
+                                                                          .catch(e => domainToSpiErrorTranslator(e)))
```

Comment:
I know you have to do it that way in order to support the SPI requirements, but maybe we should discuss it, I think it could affect the efficiency badly. (I'm not sure but let's talk about the options)